processor/src/main/java/org/mapstruct/ap/internal/model/source/Mapping.java

```diff
@@ -361,6 +374,9 @@ public Mapping copyForInheritanceTo(SourceMethod method) {
             mapping.sourceReference = sourceReference.copyForInheritanceTo( method );
         }
 
+        // TODO... must something be done here? Andreas?
+        mapping.targetReference = targetReference;
```

Comment:
That would be something related to inheriting mappings from `@MappingConfig` classes, so we should try that out in a test.